app/src/androidTest/java/org/wikipedia/model/MainView.kt

```diff
@@ -0,0 +1,22 @@
+package org.wikipedia.model
+
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.action.ViewActions
+import androidx.test.espresso.matcher.ViewMatchers
+import org.hamcrest.Matchers
+import org.wikipedia.R
+
+class MainView {
+    private val searchBarId = R.id.search_container
+
+    fun clickSearchBar(): SearchView {
+        Espresso.onView(
+            Matchers.allOf(
+                ViewMatchers.withId(searchBarId),
+                ViewMatchers.isDisplayed()
+            )
+        )
+            .perform(ViewActions.click())
+        return SearchView()
```

Comment:
Tak samo zastanawiałem się tutaj: zwracanie referencji do kolejnego page objecta - hit czy kit?

Reply:
Jak rozumiem, chciałbyś dzięki temu "chainować" metody na page'ach? Kotlin ma wbudowane tzw [scope functions](https://kotlinlang.org/docs/scope-functions.html#function-selection) które pozwalają Ci to uzyskać bez dopisywania tych returnów, więc IMO jest to zbędne. Dzięki tym funkcjom możesz mieć taki kod:
```
loginPage.run {
    typeEmail(UserCredentials.email)
    typePassword(UserCredentials.password)
    tapLoginButton()
}
```